enterprise/cmd/frontend/internal/auth/oauth/session.go

```diff
@@ -36,13 +36,14 @@ type SessionIssuerHelper interface {
 	SessionData(token *oauth2.Token) SessionData
 }
 
-func SessionIssuer(db database.DB, s SessionIssuerHelper, sessionKey string) http.Handler {
+func SessionIssuer(logger log.Logger, db database.DB, s SessionIssuerHelper, sessionKey string) http.Handler {
+	logger = logger.Scoped("SessionIssuer.HandlerFunc", "validates a token and then sets up a session")
```

Comment:
If there's the issue itself is a handler, I think just noting the scope as SessionIssuer is sufficient:

```suggestion
	logger = logger.Scoped("SessionIssuer", "validates a token and then sets up a session")
```

Reply:
Yeah good point. I was inbetween minds about this but ultimately agree!